var/spack/repos/builtin/packages/gcc/package.py

```diff
@@ -1155,5 +1155,21 @@ def runtime_constraints(cls, *, compiler, pkg):
             description=f"If any package uses %{str(compiler.spec)}, "
             f"it depends on gcc-runtime@{str(compiler.version)}:",
         )
+
+        gfortran_str = "gfortran@5"
+        if compiler.spec.satisfies("gcc@:6"):
+            gfortran_str = "gfortran@3"
+        elif compiler.spec.satisfies("gcc@7"):
+            gfortran_str = "gfortran@4"
+
+        for fortran_virtual in ("fortran-rt", gfortran_str):
+            pkg("*").depends_on(
+                fortran_virtual,
+                when=f"%{str(compiler.spec)}",
```

Comment:
What is this going to become when we no longer have the compiler attribute on the node?  How will we test whether something was "compiled" with `gcc` then?

this is a question for @becker33 too.

Reply:
When compilers are dependencies, I think this will be `when="^[virtual=fortran] gcc@X.Y"`